File: src/NodeParser/SymbolTypeNodeParser.ts
import ts from "typescript";
import { Context } from "../NodeParser.js";
import { SubNodeParser } from "../SubNodeParser.js";
import { BaseType } from "../Type/BaseType.js";
import { SymbolType } from "../Type/SymbolType.js";

export class SymbolTypeNodeParser implements SubNodeParser {
    public supportsNode(node: ts.KeywordTypeNode): boolean {
        return node.kind === ts.SyntaxKind.SymbolKeyword;
    }

    public createType(node: ts.KeywordTypeNode, context: Context): BaseType {
        return new SymbolType();
    }
}

File: internal/parser/symbol_type_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// SymbolTypeNodeParser handles the `symbol` keyword
// (src/NodeParser/SymbolTypeNodeParser.ts).
type SymbolTypeNodeParser struct{}

func NewSymbolTypeNodeParser() *SymbolTypeNodeParser {
	return &SymbolTypeNodeParser{}
}

func (p *SymbolTypeNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindSymbolKeyword
}

func (p *SymbolTypeNodeParser) CreateType(_ *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.SymbolType{}
}
